eng/pipelines/templates/steps/run-and-validate-linting.yml

```diff
@@ -61,22 +61,16 @@ steps:
   - task: PowerShell@2
     inputs:
       filePath: $(System.DefaultWorkingDirectory)/eng/pipelines/scripts/Get-Linting-Reports.ps1
-      arguments: -OutputDirectory $(Build.ArtifactStagingDirectory)/linting-reports
+      arguments: -StagingDirectory $(System.DefaultWorkingDirectory)
       pwsh: true
       workingDirectory: $(Pipeline.Workspace)
     displayName: 'Generate Linting Reports'
     continueOnError: ${{ parameters.ContinueOnError }}
     condition: or(eq(coalesce(variables['IsLatestNonLtsJdk'], true), true), and(${{ parameters.RunLinting }}, succeeded()))
 
-  - publish: $(Build.ArtifactStagingDirectory)/linting-reports.zip
-    displayName: 'Publish Linting Reports'
-    artifact: linting-reports-$(System.StageName)-$(System.JobName)-$(System.JobAttempt)
-    continueOnError: ${{ parameters.ContinueOnError }}
-    condition: or(eq(coalesce(variables['IsLatestNonLtsJdk'], true), true), and(${{ parameters.RunLinting }}, succeeded()))
-
   - pwsh: |
-      if ((Get-ChildItem -Path $(Build.ArtifactStagingDirectory)/linting-reports -Directory | Measure-Object).Count -gt 0) {
-        Write-Host "Linting errors have been found, download the 'linting-reports.zip' artifact to see what was found."
+      if (Test-Path -Path "$(System.DefaultWorkingDirectory)/troubleshooting/linting-report.zip") {
```

Comment:
TODO for myself: Have the script call above `exit 1` if there were linting errors found, we don't need two steps here.